src/configuration.ts

```diff
@@ -61,34 +64,45 @@ export class ConnectionConfiguration implements IConnectionConfiguration {
   }
 
   get puppetDir(): string {
-    return path.join(this.puppetAgentDir, 'puppet');
+    return path.join(this.puppetBaseDir, 'puppet');
   }
 
   get facterDir(): string {
-    return path.join(this.puppetAgentDir, 'facter');
+    return path.join(this.puppetBaseDir, 'facter');
   }
 
   get hieraDir(): string {
-    return path.join(this.puppetAgentDir, 'hiera');
+    return path.join(this.puppetBaseDir, 'hiera');
   }
 
   get mcoDir(): string {
-    return path.join(this.puppetAgentDir, 'mcollective');
+    return path.join(this.puppetBaseDir, 'mcollective');
   }
 
   get rubydir(): string {
-    return path.join(this.puppetAgentDir, 'sys', 'ruby');
+    switch(process.platform){
+      case 'win32':
+        return path.join(this.puppetBaseDir, 'sys', 'ruby');
+      default:
+        return path.join(this.puppetBaseDir, 'lib', 'ruby');      
```

Comment:
The vendored ruby paths are different on Mac and *nix, so use the correct path here